components/core/src/streaming_archive/writer/Archive.cpp

```diff
@@ -266,11 +266,10 @@ namespace streaming_archive { namespace writer {
     void Archive::write_msg (File& file, epochtime_t timestamp, const string& message, size_t num_uncompressed_bytes) {
         vector<encoded_variable_t> encoded_vars;
         vector<variable_dictionary_id_t> var_ids;
-        EncodedVariableInterpreter::encode_and_add_to_dictionary(message, *m_logtype_dict_entry_wrapper, m_var_dict, encoded_vars, var_ids);
+        LogTypeDictionaryEntry logtype_entry;
```

Comment:
Just realized that there is an issue.

The original code only dynamically allocates a new entry in the add occurance if the current message contains a log_str that does not exist in the hash_map, whereas this version statically creates an entry object for every message. However, I don't see a obvious runtime overhead with the newer version

I see we have 4 options 
1. keep it here since the code is cleaner and has no obivous runtime impact
2. let encode_and_add_to_dictionary returns log_value & var_positions (instead of entry_object). construct the entry object in add_occurance only if the log_value is not in the hashmap
3. reuse one entry object for all messages and clear it every time. it should work for now but i feel it could be error prone to any modificaitons we are going to make in the future
4. or we simply revert to previous version to use dynamically allocated log dictionary entry

My personal preference is 2 > 4 > 1 & 3. Please let  me know what's your opinion